Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17
+FROM golang@sha256:7dbfeb9d51c049e8bfe36cf1a4217c7b1ba304bf0eb72d57d0c04f405589f122
```

Comment:
I'm not sure where this hash is from, but it doesn't seem to match the official image for golang:1.17.

Reply:
Also, please add a comment specifying the exact version since it's not easy to tell from a hash.

Reply:
> I'm not sure where this hash is from, but it doesn't seem to match the official image for golang:1.17.

It is based on https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_digest.md
```
➜  dcrd git:(naveen/feat/fix-docker-sha) crane digest golang:1.17.0
sha256:7dbfeb9d51c049e8bfe36cf1a4217c7b1ba304bf0eb72d57d0c04f405589f122
```

Reply:
This is the way to refer to the sha256 of the image.

Reply:
Alright, I see what the issue is.  The ids on dockerhub and the ones obtained via docker pull don't match.  

For example, for golang:1.17.0 on linux/amd64, dockerhub has: https://hub.docker.com/layers/golang/library/golang/1.17.0/images/sha256-2d35a2f625aa42e8d096bcc48a4cb4df39d49440fdd9cf4d5210f42338b9492a?context=explore

Meanwhile:

```
$ docker pull golang:1.17.0
...
Digest: sha256:7dbfeb9d51c049e8bfe36cf1a4217c7b1ba304bf0eb72d57d0c04f405589f122
```

At any rate, confirmed that `7dbfeb9d51c049e8bfe36cf1a4217c7b1ba304bf0eb72d57d0c04f405589f122` is what docker pull retrieves.